R/spliceHelperFunctions.R

```diff
@@ -480,27 +484,39 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  posStrandRng <- range[which(strand == "+")]
-  negStrandRng <- range[which(strand == "-")]
-  negStrandRng <- negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))]
-  startRng <- c(posStrandRng[which(start(posStrandRng) == min(start(posStrandRng)))],
-                negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))])
-  startRng <- startRng[sort(order(startRng)[names(range)])]
-  return (startRng)
+  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
+  largestExons <- as.numeric(cumsum(elementNROWS(range)))
+  startExonsSet <- c(1, largestExons[1:length(largestExons)-1]+1)
+  startExonsSet[strand == "-"] <- largestExons[strand == "-"]
+  return(relist(unlist(range, use.names = FALSE)[startExonsSet],partitioning))
```

Comment:
you can remove the `relist()`  and `partitioning` here and return the unlisted object directly, in which case you can directly work with the unlisted object throughout the functions (works only if the expandMatchedRanges function is adopted, see comment above).

Reply:
This version takes in the unlisted startRng and endRng in the expandStartEndRanges() function.